jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -434,29 +466,46 @@ private void setModules(SchemaGeneratorConfigBuilder configBuilder) throws MojoE
      */
     private URLClassLoader getClassLoader() {
         if (this.classLoader == null) {
-            List<String> runtimeClasspathElements = null;
-            try {
-                runtimeClasspathElements = project.getRuntimeClasspathElements();
-            } catch (DependencyResolutionRequiredException e) {
-                this.getLog().error("Failed to resolve runtime classpath elements", e);
-            }
+            List<URL> urls = getUrls(Classpath.WITH_COMPILE_DEPENDENCIES);
+            this.classLoader = new URLClassLoader(urls.toArray(new URL[0]),
```

Comment:
**nitpick**: Since we know the number of URLs, might as well initialize the array with the correct length right away.
```suggestion
            this.classLoader = new URLClassLoader(urls.toArray(new URL[urls.size()]),
```

Reply:
Changed that